mirrord/layer/src/lib.rs

```diff
@@ -1020,14 +458,28 @@ pub(crate) unsafe extern "C" fn close_detour(fd: c_int) -> c_int {
 /// on macOS, be wary what we do in this path as we might trigger https://github.com/metalbear-co/mirrord/issues/1745
 #[hook_guard_fn]
 pub(crate) unsafe extern "C" fn fork_detour() -> pid_t {
-    debug!("Process {} forking!.", std::process::id());
+    tracing::debug!("Process {} forking!.", std::process::id());
+
+    let parent_connection = PROXY_CONNECTION.get().expect("PROXY_CONNECTION not set");
+    let new_connection = ProxyConnection::new(
+        parent_connection.proxy_addr(),
+        NewSessionRequest::Forked(parent_connection.layer_id()),
+        Duration::from_secs(5),
+    )
+    .expect("failed to establish proxy connection for child");
+
     let res = FN_FORK();
     if res == 0 {
-        debug!("Child process initializing layer.");
+        tracing::debug!("Child process initializing layer.");
+        PROXY_CONNECTION
+            .set(new_connection)
+            .expect("setting PROXY_CONNECTION");
         mirrord_layer_entry_point()
     } else {
-        debug!("Child process id is {res}.");
+        tracing::debug!("Child process id is {res}.");
+        std::mem::drop(new_connection);
```

Comment:
After consideration - replaced explicit `drop` with a comment. `new_connection` is dropped at the end of the scope anyway. By the way, fixed a bug - `fork` was not checked for errors

Reply:
> By the way, fixed a bug - `fork` was not checked for errors

Was the bug that the the error code would be displayed in the debug log? Otherwise I don't see it.

Reply:
Yup, that's it